backend/src/LibExecution/ProgramTypes.fs

```diff
@@ -396,19 +394,19 @@ type Expr =
   | EInfix of id * Infix * Expr * Expr
   // the id in the varname list is the analysis id, used to get a livevalue
   // from the analysis engine
-  | ELambda of id * List<id * string> * Expr
+  | ELambda of id * NEList<id * string> * Expr
   | EFieldAccess of id * Expr * string
   | EVariable of id * string
   | EApply of id * Expr * typeArgs : List<TypeReference> * args : List<Expr>
   | EFnName of id * NameResolution<FnName.T>
   | EList of id * List<Expr>
   | EDict of id * List<string * Expr>
   | ETuple of id * Expr * Expr * List<Expr>
-  | EPipe of id * Expr * PipeExpr * List<PipeExpr>
+  | EPipe of id * Expr * NEList<PipeExpr>
 
   // See NameResolution comment above
-  | ERecord of id * NameResolution<TypeName.T> * List<string * Expr>
-  | ERecordUpdate of id * record : Expr * updates : List<string * Expr>
+  | ERecord of id * NameResolution<TypeName.T> * NEList<string * Expr>
+  | ERecordUpdate of id * record : Expr * updates : NEList<string * Expr>
```

Comment:
Suppose the user types:

```fsharp
{ myRecord with /* nothing */ }
```

Struggling to make any sense of this, so NEList feels OK,